smaug/operators/common.h

```diff
@@ -176,58 +271,109 @@ typedef unsigned ufx_t;
 typedef uint16_t fp16_t;
 typedef uint16_t float16;
 
+#define CACHELINE_SIZE 32
+#define LOG_PAGE_SIZE 12
+
+/** \defgroup VectorData
```

Comment:
It'd great if we have a docs repo so I can get to see what this looks like.

Reply:
Yeah I agree :) but let me finish a rough first version of all the documentation first while I'm in the zone, then we can do the rest.